src/analysis/normed_space/star/complex.lean

```diff
@@ -0,0 +1,77 @@
+/-
+Copyright (c) 2022 Frédéric Dupuis. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Frédéric Dupuis
+-/
+
+import analysis.normed_space.star.basic
+import algebra.star.self_adjoint
+import analysis.complex.basic
+
+/-!
+# Complex normed star modules and algebras
+
+Facts about star modules and star algebras over the complex numbers.
+
+## Main definitions
+
+* `star_ring.re`: the real part of an element of a star module, defined as `2⁻¹ • (x + star x)`
+* `star_ring.im`: the imaginary part of an element of a star module, defined as
+  `(-I * 2⁻¹) • (x - star x)`.
+
+-/
+
+variables {E : Type*}
+
+namespace star_module
+open_locale complex_conjugate
+open complex
+
+variables [add_comm_group E] [star_add_monoid E] [module ℂ E] [star_module ℂ E]
+
+/-- The real part of an element of a star module, as a real-linear map. -/
+@[simps] noncomputable def re : E →ₗ[ℝ] self_adjoint E :=
+{ to_fun := λ x, ⟨(2⁻¹ : ℂ) • (x + star x), by simp only [self_adjoint.mem_iff, star_smul, add_comm,
+                                        star_add_monoid.star_add, star_inv', star_bit0,
+                                        star_one, star_star]⟩,
+  map_add' := λ x y, by { ext, simp [add_add_add_comm] },
+  map_smul' := λ r x, by { ext, simp [smul_comm (2⁻¹ : ℂ) r] } }
```

Comment:
Does this generalize nicely to elements of arbitrary star modules with `invertible 2`? I'm thinking something like `re (w + xi + yj + zk) = w` on the quaternions, and maybe `re m` on a matrix giving the symmetric part

Reply:
That ends up looking something like:
```lean
def star.re (R) [semiring R] [invertible (2 : R)] [star_monoid R] [has_trivial_star R]
  [add_comm_group E] [module R E] [star_add_monoid E] [star_module R E] :
  E →ₗ[R] self_adjoint E :=
{ to_fun := λ x, ⟨⅟(2 : R) • (x + star x),
                          by {
                            simp only [self_adjoint.mem_iff, star_smul, add_comm,
                                        star_add_monoid.star_add, star_inv_of (2 : R), star_bit0,
                                        star_one, star_star],
                            sorry
                          }⟩,
  map_add' := λ x y, by simp [add_add_add_comm],
  map_smul' := λ r x,
    have ⅟ 2 * r = r * ⅟ 2, from commute.inv_of_left (commute.one_left r).bit0_left,
    by simp [←mul_smul, this] }
```
which depends on #11951. The sorry is trivial but I can't work out how to convince lean of it.

Reply:
I'm not sure whether trying to generalize `im` makes as much sense - the "imaginary" part of the quaternions does not live in the same space as the real part.

Reply:
Right, I left `im` in the original file, only for the complex case.